src/HomeScreen.tsx

```diff
@@ -1,29 +1,54 @@
-import React, { Component } from "react";
+import React, { Component } from 'react';
 
-import { VoteButton } from "shared";
+import { VoteButton } from 'shared';
 
 // TODO: smh think of a better name lol
-const BoxInput = ({title, iconClassname}: {title: string, iconClassname: string}) => (
-  <button className="btn home-screen__box-input" onClick={() => alert(title)}>
-      <div className={iconClassname}/>
-      <div className="type--large-white">{title}<span className="type--large-green">.</span></div> 
-    </button>
+const BoxInput = ({
+  title,
+  iconClassname,
+  onClickHandler
+}: {
+  title: string;
+  iconClassname: string;
+  onClickHandler: (title: string) => boolean;
+}) => (
+  <button
+    className="btn home-screen__box-input"
+    onClick={() => onClickHandler(title)}
+  >
+    <div className={iconClassname} />
+    <div className="type--large-white">
+      {title}
+      <span className="type--large-green">.</span>
+    </div>
+  </button>
 );
 
-const HomeScreen = () => (
+interface HomeScreen {
+  onClickHandler: (title: string) => boolean;
+}
+const HomeScreen = ({ onClickHandler }: HomeScreen) => (
```

Comment:
So is homeScreen gonna be interacting with the data layer at all, or is this gonna be all presentational?

Reply:
hmmm I think we'll be dispatching an action to create a new room or join a new room from this component - so we will end up connecting it I think. Does that answer your question?